File: frontend/src/api/report.ts
/**
 * Echo Mock System - 报告 API 请求
 */

import request from '../utils/request'

export interface DimensionScore {
  name: string
  score: number
}

export interface QaEvaluationItem {
  id: string
  question_content: string
  user_answer: string | null
  audio_analysis_status: string
  speech_rate: number | null
  pause_ratio: number | null
  technical_score: number | null
  correction_feedback: string | null
}

export interface ReportResponse {
  session_id: string
  target_role: string
  status: string
  overall_score: number | null
  dimension_scores: DimensionScore[] | null
  comprehensive_report: string | null
  evaluations: QaEvaluationItem[]
}

/** 获取面试综合报告 */
export const getReport = (sessionId: string) =>
  request.get<any, ReportResponse>(`/reports/${sessionId}`)

File: backend/app/schemas/report.py
"""
Echo Mock System - 报告相关 Pydantic 模型
"""

from typing import Any, Dict, List, Optional

from pydantic import BaseModel, Field


class DimensionScore(BaseModel):
    """单维度评分项 - 供 ECharts 雷达图消费"""
    name: str = Field(..., description="维度名称（如：技术深度、逻辑表达、沟通能力）")
    score: float = Field(..., ge=0, le=100, description="该维度得分（0-100）")


class QaEvaluationItem(BaseModel):
    """单道问答的评估明细"""
    id: str
    question_content: str
    user_answer: Optional[str] = None
    audio_analysis_status: str
    speech_rate: Optional[float] = None
    pause_ratio: Optional[float] = None
    technical_score: Optional[float] = None
    correction_feedback: Optional[str] = None

    class Config:
        from_attributes = True


class ReportResponse(BaseModel):
    """完整的面试报告响应"""
    session_id: str
    target_role: str
    status: str
    overall_score: Optional[float] = None
    dimension_scores: Optional[List[DimensionScore]] = None
    comprehensive_report: Optional[str] = None
    evaluations: List[QaEvaluationItem] = []
